tasks/main.yml

```diff
@@ -46,11 +46,10 @@
 
 - name: Register agent service status
   shell: |
-    if [ ! -f "/etc/systemd/system/$(systemd-escape vsts.agent.{{ az_devops_accountname }}.{{ az_devops_agent_pool_name }}.{{ az_devops_agent_name }}.service)" ]; then
-      true
-    else
-      false
-    fi
+    for f in /etc/systemd/system/vsts.agent.$(systemd-escape {{ az_devops_accountname }} | sed 's/\\/\\\\/g').*.$(systemd-escape {{ az_devops_agent_name }} | sed 's/\\/\\\\/g').service"; do
+        [ -e "$f" ] && exit 1 || exit 0
+        break
+    done
```

Comment:
I don't understand this live @TJEvans .  Could you please elaborate why the for loop?
Thanks.

Reply:
Wildcarding the agent pool makes this a potential list return.  I ran into the issue describen in https://github.com/gsoft-inc/ansible-role-azure-devops-agent/issues/18 when I was trying to move an existing agent from the default to named pool.